src/main/java/com/epam/esm/integration/sqlrepo/MySQLRepository.java

```diff
@@ -0,0 +1,243 @@
+import java.sql.ResultSet;
+import java.sql.Timestamp;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.*;
+
+@Repository
+public class MySQLRepository implements TagsRepository, GiftCertificatesRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final TransactionTemplate transactionTemplate;
+    private final KeyHolder keyHolder;
+
+    @Autowired
+    public MySQLRepository(JdbcTemplate jdbcTemplate, TransactionTemplate transactionTemplate, KeyHolder keyHolder) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.transactionTemplate = transactionTemplate;
+        this.keyHolder = keyHolder;
+    }
+
+    @Override
+    public Optional<Tag> getTagById(Long id) {
+        return jdbcTemplate.query(SQLQuery.SELECT_FROM_TAGS_WHERE_ID, rs -> {
+            Tag tag = null;
+            if (rs.next()) {
+                tag = new Tag(rs.getLong("id"),
+                        rs.getString("name"));
+            }
+            return tag == null ? Optional.empty() : Optional.of(tag);
+        }, id);
+    }
+
+    @Override
+    public Optional<Tag> createNewTag(Tag newTag) {
+        return transactionTemplate.execute(status -> {
+                    createTagsInTransaction(keyHolder, newTag);
+                    return getTagById(Objects.requireNonNull(keyHolder.getKey()).longValue());
+                }
+        );
+    }
+
+    @Override
+    public boolean deleteTagById(Long id) {
+        return jdbcTemplate.update(SQLQuery.DELETE_FROM_TAGS_WHERE_ID, id) == 1;
+    }
+
+    @Override
+    public boolean tagByNameExist(String name) {
+        Integer exist = jdbcTemplate.queryForObject(SQLQuery.SELECT_FROM_TAGS_BY_NAME,Integer.class,name);
+        if (exist == null) return false;
+        else return exist>0;
+    }
+
+    @Override
+    public Optional<List<Tag>> getAllTags() {
+        return Optional.of(jdbcTemplate.query(SQLQuery.SELECT_FROM_TAGS, (ResultSet resultset, int i) ->
+                new Tag(resultset.getLong("id"),
+                        resultset.getString("name"))));
+    }
+
+    @Override
+    public Optional<GiftCertificate> getGiftCertificateById(Long id) {
+        return jdbcTemplate.query(SQLQuery.SELECT_FROM_CERTIFICATES_WITH_TAGS_BY_ID, rs -> {
+            GiftCertificate giftCertificate = null;
+            while (rs.next()) {
+                if (giftCertificate == null) {
+                    giftCertificate = new GiftCertificate(
+                            rs.getLong("id"),
+                            rs.getString("name"),
+                            rs.getString("description"),
+                            rs.getBigDecimal("price"),
+                            Duration.ofDays(rs.getLong("duration")).toString(),
+                            rs.getTimestamp("create_date").toLocalDateTime().toString(),
+                            rs.getTimestamp("last_update_date").toLocalDateTime().toString(),
+                            null);
```

Comment:
What is this null. Create a variable with name and pass here for readability.
e.g.
String optionalDescription = null
new GiftCertificate(..... , optionalDescription);